aibolit/patterns/method_chaining/method_chaining.py

```diff
@@ -0,0 +1,75 @@
+import javalang
+
+import uuid
+from collections import defaultdict
+
+
+class MethodChainFind:
+
+    def __init__(self):
+        pass
+
+    def traverse_node(self, node, dict_with_chains, uuid_method):
+        if not node:
+            return dict_with_chains
+
+        for item in node.children:
+            if item and (isinstance(item, tuple) or isinstance(item, list)):
+                for j in item:
+                    if isinstance(j, javalang.tree.MethodInvocation):
+                        if not j.qualifier and j.qualifier != '':
+                            # it means that there is method chaining
+                            dict_with_chains[uuid_method].append([j.position.line, j.member])
+                            self.traverse_node(j, dict_with_chains, uuid_method)
+                        else:
+                            # it means that we have method invocation without chaining like
+                            # result.add(field.getName(), column.columnName(), field.getType());
+                            new_uuid = uuid.uuid1()
+                            dict_with_chains[new_uuid].append([j.position.line, j.member])
+                            self.traverse_node(j, dict_with_chains, new_uuid)
+
+                    elif isinstance(j, javalang.tree.MethodDeclaration):
+                        self.traverse_node(j, dict_with_chains, str(uuid.uuid1()))
+
+                    elif isinstance(j, javalang.tree.StatementExpression):
+                        self.traverse_node(j, dict_with_chains, uuid_method)
+
+                    elif isinstance(j, javalang.tree.This) or isinstance(j, javalang.tree.ClassCreator):
+                        self.traverse_node(j, dict_with_chains, str(uuid.uuid1()))
+            elif isinstance(item, javalang.tree.ClassCreator):
+                self.traverse_node(item, dict_with_chains, uuid_method)
+
+        return dict_with_chains
+
+    def __file_to_ast(self, filename: str) -> javalang.ast.Node:
```

Comment:
@lyriccoder This method should be extracted to a new class / interface. So, you would pass the file you want to have the AST extracted and we would get it in the `value` method of this new class
´